dashboards/clicks_and_impressions.py

```diff
@@ -0,0 +1,92 @@
+import streamlit as st
+import matplotlib.pyplot as plt
+
+from utils.duckdb import get_dbcur
+
+st.title("Clicks & Impressions Dashboard")
+
+cur = get_dbcur()
+
+query = """
+WITH ranked_events AS (
+    SELECT 
+        ed.website_event_id,
+        es.id AS mentor_id,
+        ANY_VALUE(es.name) AS mentor_name,
+        SUM(CASE WHEN we.event_name = 'Click' THEN 1 ELSE 0 END) AS click_count,
+        SUM(CASE WHEN we.event_name = 'Impression' THEN 1 ELSE 0 END) AS impression_count
+    FROM 
+        umamidb.event_data ed
+    JOIN 
+        umamidb.website_event we
+        ON ed.website_event_id = we.event_id
+    JOIN 
+        memory.elasticsearch es
+        ON es.id = ed.string_value
+    WHERE 
+        (we.event_name = 'Click' OR we.event_name = 'Impression')
+        AND we.url_path LIKE '/mentors%'
+        AND we.referrer_path LIKE '/mentors%'
```

Comment:
These two criteria shouldn't be necessary, as the click/impression events can only occur on `/mentors%`. (Also, note that `referrer_path` is the previous URL visited by the user, so it should have no bearing on the tracking of click/impression events.)

See the following query which is empty:
![image](https://github.com/user-attachments/assets/76634754-dff6-43cb-9afc-1d1b01c8364a)